src/synchronizeContacts.js

```diff
@@ -1,30 +1,202 @@
-const synchronizeContacts = (
-  localContacts,
-  remoteContacts,
-  synchronizationStrategy
+const get = require('lodash/get')
+const union = require('lodash/union')
+
+const { DOCTYPE_CONTACTS, APP_NAME } = require('./constants')
+const getInitialMetadata = require('./getInitialMetadata')
+const mergeContact = require('./mergeContact')
+const transpiler = require('./transpiler')
+
+const findGoogleContact = (cozyContact, googleContacts, contactAccountId) => {
+  const resourceName = get(
+    cozyContact,
+    ['cozyMetadata', 'sync', contactAccountId, 'id'],
+    null
+  )
+
+  if (resourceName) {
+    return googleContacts.find(
+      googleContact => googleContact.resourceName === resourceName
+    )
+  }
+
+  return undefined
+}
+
+const findCozyContact = (googleContact, cozyContacts, contactAccountId) => {
+  return cozyContacts.find(
+    contact =>
+      get(contact, ['cozyMetadata', 'sync', contactAccountId, 'id']) ===
+      googleContact.resourceName
+  )
+}
+
+const updateCozyMetadata = (
+  cozyContact,
+  etag,
+  resourceName,
+  contactAccountId
+) => {
+  const now = new Date().toISOString()
+  const updatedByApps = union(
+    get(cozyContact, 'cozyMetadata.updatedByApps', []),
+    [APP_NAME]
+  )
+
+  return {
+    ...cozyContact,
+    cozyMetadata: {
+      ...cozyContact.cozyMetadata,
+      updatedAt: now,
+      updatedByApps,
+      sync: {
+        [contactAccountId]: {
+          konnector: APP_NAME,
+          lastSync: now,
+          remoteRev: etag,
+          id: resourceName,
+          contactsAccountsId: contactAccountId
+        }
+      }
+    }
+  }
+}
```

Comment:
Seems to be very close to https://github.com/konnectors/cozy-konnector-google/blob/e0755fcce9140885452d64f23cc6ac415582dbaa/src/getInitialMetadata.js#L12 

Do we need to create some kind of abstraction to deal with cozyMeta ?

Reply:
As in the end cozy-client will write these metadata (expect sync info), we think it probably does not worth it right now. You had something particular in mind for this case?